pkg/resolution/common/interface.go

```diff
@@ -45,6 +45,7 @@ type Request interface {
 	Name() string
 	Namespace() string
 	Params() pipelinev1.Params
+	Url() string
```

Comment:
bleh this is also going to be a breaking change, but one that's going to be difficult to avoid :(

If we want to be paranoid this needs to be another interface, or at the very least make sure there's a breaking change warning.

Reply:
Would that be the case every time we add a new field? 
I'm wondering how we can continue to update the interface without breaking users.

I think this interface is used internally by us. Users using the framework we provide won't be affected by this I think. But obviously, since this is exported, users could be relying on this.

Reply:
Struct fields no, interfaces yes.

The backwards compatible way to make this change would be to break this into separate interfaces, and have the consuming implementations satisfy both.

e.g.


```go
type Request interface {
	Name() string
	Namespace() string
	Params() pipelinev1.Params
}

type URLRequest interface {
	URL() string
}

...

type myRequestImpl struct {
	Request
	URLRequest
}
```

Or alternatively if you wanted to do an either-or approach:

```go
type NamespacedRequest interface {
	Name() string
	Namespace() string
}

type Request interface {
	NamespacedRequest
	Params() pipelinev1.Params
}

type URLRequest interface {
	NamespacedRequest
	URL() string
}
```

Reply:
If this is indeed internal only, then we should unexport / move to an internal package (which would also be a breaking change, but if we're sure about this then it's probably worth doing sooner rather than later)

Reply:
Users only rely on the framework interface: https://tekton.dev/docs/pipelines/how-to-write-a-resolver/

Reply:
[I tried refactoring this](https://github.com/tektoncd/pipeline/pull/7910). But since this resolution is now already Stable, refactoring it cleanly and confidently is not possible.

I'm not too paranoid of breaking this specific interface so I'm thinking of leaving a breaking message in the go-docs. However, the other interface in `pkg/framework` is directly user facing so we can add a new interface there like you suggested in the above comment.